cmd/kn/main.go

```diff
@@ -106,46 +111,12 @@ func run(args []string) error {
 	}
 }
 
-// Get only the args provided but no options. The extraction
-// process is a bit tricky as Cobra doesn't provide such
-// functionality out of the box
-func stripFlags(args []string) ([]string, error) {
-	// Store all command
-	commandsFound := &[]string{}
-
-	// Use a canary command that allows all options and only extracts
-	// commands. Doesn't work with arbitrary boolean flags but is good enough
-	// for us here
-	extractCommand := cobra.Command{
-		Run: func(cmd *cobra.Command, args []string) {
-			for _, arg := range args {
-				*commandsFound = append(*commandsFound, arg)
-			}
-		},
-		SilenceErrors: true,
-		SilenceUsage:  true,
-	}
-
-	// No help and usage functions to prin
-	extractCommand.SetHelpFunc(func(*cobra.Command, []string) {})
-	extractCommand.SetUsageFunc(func(*cobra.Command) error { return nil })
-
-	// Filter out --help and -h options to avoid special treatment which we don't
-	// need here
-	extractCommand.SetArgs(filterHelpOptions(args))
-
-	// Adding all global flags here
-	config.AddBootstrapFlags(extractCommand.Flags())
-
-	// Allow all options
-	extractCommand.FParseErrWhitelist = cobra.FParseErrWhitelist{UnknownFlags: true}
-
-	// Execute to get to the command args
-	err := extractCommand.Execute()
-	if err != nil {
-		return nil, err
+// Get only the args provided but no options
+func stripFlags(rootCmd *cobra.Command, args []string) ([]string, error) {
+	if err := rootCmd.ParseFlags(filterHelpOptions(args)); err != nil {
+		fmt.Errorf("error while parsing flags from args %v: %s", args, err.Error())
```

Comment:
Do you need a return statement here ? What's the usage of the error object created by `fmt.Errorf` ? I guess maybe you want to:
```
return nil, fmt.Errorf("error while parsing flags from args %v: %s", args, err.Error())
```

Reply:
good catch, fixed.